### PR TITLE
Fix initialization in range iSAM example

### LINKED ISSUE #1130: RangeISAMExample Error 'gtsam::ValuesKeyDoesNotExist'
I am fairly new to GTSAM and am trying to run the RangeISAMExample.cpp. When I run the code as is I get the following error:
`[100%] Built target RangeISAMExample_plaza2
terminate called after throwing an instance of 'gtsam::ValuesKeyDoesNotExist'
  what():  Attempting to at the key "L2", which does not exist in the Values.
Aborted (core dumped)
make[3]: *** [examples/CMakeFiles/RangeISAMExample_plaza2.run.dir/build.make:57: examples/CMakeFiles/RangeISAMExample_plaza2.run] Error 134
make[2]: *** [CMakeFiles/Makefile2:21082: examples/CMakeFiles/RangeISAMExample_plaza2.run.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:21089: examples/CMakeFiles/RangeISAMExample_plaza2.run.dir/rule] Error 2
make: *** [Makefile:8556: RangeISAMExample_plaza2.run] Error 2`

### PR DESCRIPTION
**Pull request by dellaert**
I cleaned up this old example and initialized the landmarks randomly as iSAM proceeds.
Fixes #1130

### CODE REVIEW

**Comment by jerredchen** [comment]
@dellaert I took at look at the issue #1130 and when I built the branch, I'm actually running into the same issue:
```
terminate called after throwing an instance of 'gtsam::IndeterminantLinearSystemException'
  what():  
Indeterminant linear system detected while working near variable
3086 (Symbol: 3086).
```
I'm still going through the code and haven't found anything in particular.

**Comment by dellaert** [comment]
> @dellaert I took at look at the issue #1130 and when I built the branch, I'm actually running into the same issue:
> 
> ```
> terminate called after throwing an instance of 'gtsam::IndeterminantLinearSystemException'
>   what():  
> Indeterminant linear system detected while working near variable
> 3086 (Symbol: 3086).
> ```
> 
> I'm still going through the code and haven't found anything in particular. My understanding is that if you're not getting an ill-posed solution on your Mac, there might possibly be OS differences (?) - I'm not sure though.

What OS are you on? Is it also on L3?

**Comment by jerredchen** [comment]
@dellaert I'm on Ubuntu 20.04 (not sure what you mean by L3).

**Comment by dellaert** [comment]
> @dellaert I'm on Ubuntu 20.04 (not sure what you mean by L3).

The exception refers to a variable. What is it?

**Comment by dellaert** [comment]
Oh wait, just saw: it's pose 3086, not 3.

**Comment by jerredchen** [comment]
Right, just realized that L3 referred to the landmark symbol. For my error it's actually not occurring with a landmark variable:
```
terminate called after throwing an instance of 'gtsam::IndeterminantLinearSystemException'
  what():  
Indeterminant linear system detected while working near variable
3086 (Symbol: 3086).
```

**Comment by dellaert** [comment]
Thanks!! Can I trouble you to try to set the standard deviation for initializing the landmark to something less than 1000? Try 100, 10. Check several random seeds for each setting, and see whether 10 or 100 reliably succeeds. It could be a numeric issue in computing the determinant - should not be the case but that would be another issue to investigate.

**Comment by varunagrawal** [comment]
The Windows CI is failing because the scoop changed its default install process. It annoys the heck out of me that the repo and the build commands are so intertwined.
Anyway, my last PR has the fix for it, so please merge in develop.

**Comment by dellaert** [comment]
> Okay sure, I'll let you know if I find anything interesting that comes up.

Ping @jerredchen ?

**Comment by jerredchen** [comment]
@dellaert Even with changing the loose prior sigma and trying different seeds, it still states that the system is ill-posed. I've only had success with one set of parameters, where `looseNoise = NM::Isotropic::Sigma(2, 1)` and `std::normal_distribution<double> normal(0.0, 10)`, which will yield:
```
-Total: 0 CPU (0 times, 0 wall, 49.39 children, min: 0 max: 0)
|   -iSAM: 49.49 CPU (1 times, 28.1184 wall, 49.39 children, min: 0 max: 0)
|   |   -batchInitialization: 1.04 CPU (1 times, 0.229335 wall, 1.04 children, min: 0 max: 0)
|   |   -update: 26.38 CPU (82 times, 5.82597 wall, 26.38 children, min: 0 max: 0)
|   |   -calculateEstimate: 21.97 CPU (82 times, 21.9526 wall, 21.97 children, min: 0 max: 0)
L0:-0.58908 0.627698
L1:-0.867836  0.919635
L2:-2.54579  2.81404
L3: 6.26313 -6.91674
L4: -0.2766 0.303318
L5:-0.888326  0.974642
L6:-0.784498  0.899885
L7:-0.264646  0.220543
L8:-0.174773  0.242514
L9:-0.0343196  0.0533935
```
But trying different random seeds with the same parameters will still result in a supposedly underdetermined system so I could only conclude that this worked by coincidence

**Comment by dellaert** [comment]
Update, after all this, it turned out to be a *parsing* error :-( The TD file has floats in it, and one of them was parsed as an int, so the data association was all messed up. I figured this out only after adding a python notebook (now also part of this PR) where everything worked nicely as I used `np.loadtxt`. Result in there looks like:

image

Which looks exactly like trajectory in https://www.ri.cmu.edu/pub_files/2009/9/Final_5datasetsRangingRadios.pdf.

Note, the range noise has to be made artificially high for iSAM to succeed. To get good uncertainties in line with the UWB standard deviation of 0.5 m (see the paper) a fine-tuning step would be needed. Or, SmartRangeFactor can be used as well.

**Comment by dellaert** [comment]
@jerredchen check on Linux then approve and merge if it works?

**Comment by jerredchen** [comment]
@dellaert Sounds great, I will build the changes and if everything works and the code looks fine then I will merge.

**Comment by jerredchen** [comment]
@dellaert Tested on Linux - the landmark locations of the C++ and the notebook versions match and look good. Other than the small suggestion I made, it looks good to merge.

**Comment by dellaert** [comment]
@jerredchen I don't see a suggestion. Did you use the review functionality but forgot to hit approve? If agree with your suggestion I'll edit before merging (but I can't merge without an approve).

**Review by jerredchen — APPROVED** [review]

**Comment by jerredchen** [comment]
@dellaert You're right, I completely forgot to submit the review. It's a really small thing, so other than that I think it's good to be merged.

File: examples/RangeISAMExample_plaza2.cpp
Comment (dellaert) [reply examples/RangeISAMExample_plaza2.cpp:41]: It's a bit confusing but I'm not gonna restart the CI for this :-) CO2 emissions and all that. I'll slip it into the other PR that I'm working on.

**Pull request merged**
